src/main/java/com/google/firebase/auth/UserRecord.java

```diff
@@ -177,6 +191,14 @@ private static void checkEmail(String email) {
     checkArgument(email.matches("^[^@]+@[^@]+$"));
   }
 
+  private static void checkPhoneNumber(String phoneNumber) {
+    // Phone number verification is very lax here. Backend will enforce E.164 spec compliance, and
+    // normalize accordingly.
+    checkArgument(!Strings.isNullOrEmpty(phoneNumber), "phone number cannot be null or empty");
+    checkState(phoneNumber.matches("^\\+.*[a-zA-Z0-9].*$"),
```

Comment:
I simplified this to check only for the `+` prefix. Updated the error message accordingly. We can rely on the backend server for more comprehensive validation of phone numbers.